scripts/benchmark.py

```diff
@@ -0,0 +1,192 @@
+#!/usr/bin/python3
+import argparse
+import json
+import os
+import statistics
+import subprocess
+import sys
+
+from monotonic import monotonic
+
+from detect_secrets.core.usage import PluginOptions
+
+
+def main():
+    args = get_arguments()
+
+    print(
+        'Running performance tests on: {}'.format(
+            ', '.join(args.plugin),
+        ),
+        file=sys.stderr,
+    )
+    print(
+        'for: {}'.format(args.filenames),
+        file=sys.stderr,
+    )
+
+    # First, convert chosen plugins into their disabled flags
+    always_disabled_plugins = []
+    flag_list = {}
+    for info in PluginOptions.all_plugins:
+        if info.classname in args.plugin:
+            flag_list[info.disable_flag_text] = info.classname
+        else:
+            always_disabled_plugins.append(info.disable_flag_text)
+
+    # Then, iterate through each disabled flag, toggling them off
+    # individually.
+    timings = {}
+    if len(args.plugin) == len(PluginOptions.all_plugins):
+        # Only run benchmarks for all the cases, if already running all plugins
+        timings['all-plugins'] = time_execution(
+            filenames=args.filenames,
+            timeout=args.harakiri,
+            num_iterations=args.num_iterations,
+        )
+
+    for flag_number, flag in enumerate(flag_list):
+        plugins_to_ignore = list(flag_list.keys())
+        plugins_to_ignore.pop(flag_number)
+
+        key = flag_list[flag]
+        timings[key] = time_execution(
+            filenames=args.filenames,
+            timeout=args.harakiri,
+            num_iterations=args.num_iterations,
+            flags=plugins_to_ignore + always_disabled_plugins,
+        )
+
+    print_output(timings, args)
+
+
+def get_arguments():
+    plugins = [
+        info.classname
+        for info in PluginOptions.all_plugins
+    ]
+
+    parser = argparse.ArgumentParser(description='Run some benchmarks.')
+    parser.add_argument(
+        'filenames',
+        nargs=argparse.REMAINDER,
+        help='Filenames to check or detect-secrets compatible arguments.',
+    )
+    parser.add_argument(
+        '--pretty',
+        action='store_true',
+        help='Human readable output.',
+    )
+    parser.add_argument(
+        '--plugin',
+        default=None,       # needs to be None, otherwise append won't work as expected
+        choices=plugins,
+        action='append',
+        help=(
+            'Specifies a plugin to test. May provide multiple values. '
+            'Defaults to all.'
+        ),
+    )
+    parser.add_argument(
+        '--harakiri',
```

Comment:
lol

Reply:
FWIW, I copied uwsgi's syntax for this: https://uwsgi-docs.readthedocs.io/en/latest/Glossary.html